docs/releases/5.1.txt

```diff
@@ -91,12 +91,15 @@ redirects all unauthenticated requests to a login page. Views can allow
 unauthenticated requests by using the new
 :func:`~django.contrib.auth.decorators.login_not_required` decorator.
 
-The :class:`~django.contrib.auth.middleware.LoginRequiredMiddleware` respects
-the ``login_url`` and ``redirect_field_name`` values set via the
+``LoginRequiredMiddleware`` respects the ``login_url`` and
+``redirect_field_name`` values set via the
 :func:`~.django.contrib.auth.decorators.login_required` decorator, but does not
 support setting ``login_url`` or ``redirect_field_name`` via the
 :class:`~django.contrib.auth.mixins.LoginRequiredMixin`.
 
+``LoginRequiredMiddleware`` has not been added to the ``startproject``
+template—projects need to opt-in by adding it to :setting:`MIDDLEWARE`.
```

Comment:
How about...
```suggestion
To enable this middleware you need to add
:class:`django.contrib.auth.middleware.LoginRequiredMiddleware` to the
:setting:`MIDDLEWARE` setting **after** the
:class:`django.contrib.auth.middleware.AuthenticationMiddleware`::

    MIDDLEWARE = [
        "...",
        "django.contrib.auth.middleware.AuthenticationMiddleware",
        "django.contrib.auth.middleware.LoginRequiredMiddleware",
        "...",
    ]
```
([inspired by RemoteUserMiddleware How-to](https://docs.djangoproject.com/en/5.0/howto/auth-remote-user/#configuration))

Reply:
I think that’s too much detail for the release notes, so I’ll instead add this to the reference docs.

Reply:
```suggestion
To enable this, add ``"django.contrib.auth.middleware.LoginRequiredMiddleware"``
to your :setting:`MIDDLEWARE` setting.
```
I wouldn't mention the start project template project as some people may not know it exists